src/main/kotlin/uk/gov/dluhc/printapi/service/aed/AnonymousElectorDocumentSearchService.kt

```diff
@@ -23,16 +26,36 @@ class AnonymousElectorDocumentSearchService(
     fun searchAnonymousElectorDocumentSummaries(
         dto: AnonymousSearchCriteriaDto
     ): AnonymousSearchSummaryResults {
+        val gssCodes = eroService.lookupGssCodesForEro(dto.eroId)
+        with(getAedPagedSummaries(dto = dto, gssCodes = gssCodes)) {
+            return AnonymousSearchSummaryResults(
+                results = content
+            )
+        }
+    }
+
+    private fun getAedPagedSummaries(
+        dto: AnonymousSearchCriteriaDto,
+        gssCodes: List<String>
+    ): Page<AnonymousSearchSummaryDto> {
         with(dto) {
-            val gssCodes = eroService.lookupGssCodesForEro(eroId)
-            val pagedSummaries = anonymousElectorDocumentSummaryRepository
-                .findAllByGssCodeInAndSourceType(
-                    gssCodes = gssCodes,
-                    sourceType = ANONYMOUS_ELECTOR_DOCUMENT,
-                    pageRequest = buildPageRequest(page = page, pageSize = pageSize)
-                ).map { anonymousSearchSummaryMapper.toAnonymousSearchSummaryDto(it) }
+            val pageRequest = buildPageRequest(page = page, pageSize = pageSize)
+            return when (searchBy) {
```

Comment:
Minor (and I'm not asking you to change this), but this is probably OK for now based on current requirements, but if we ever get asked to add more search by parameters this is going to get hard to manage.
Its OK for now, but if we get asked to add another searchBy parameter I think we'll probably want to refactor this, and do something similar to VCA in respect of specification builder etc